megamek/src/megamek/client/ui/swing/RandomMapDialog.java

```diff
@@ -414,6 +428,17 @@ private boolean doApply() {
         MAP_SETTINGS_OBSERVER.updateMapSettings(newMapSettings);
         return true;
     }
+    
+    public boolean activateDialog() {
+    	userCancel = false;
+    	setVisible(true);
+    	return userCancel;
```

Comment:
This seems kind of pointless - you're setting userCancel to false then returning it without any possibility of changing it.

Reply:
This is a modal dialog. setVisible does not return until the dialog is closed.